src/invoke.cpp

```diff
@@ -3794,19 +3794,32 @@ static bool copy_mat(lk::invoke_t &cxt, wxString sched_name, matrix_t<double> &m
 	return true;
 }
 
-void fcall_geocode(lk::invoke_t& cxt) 
+void fcall_geocode(lk::invoke_t& cxt)
 {
 	LK_DOC("geocode",
-		"Given a street address or location name, returns latitude, longitude, and time zone. Not designed to take latitude and longitude as input. Uses the MapQuest Geocoding API via a private NREL wrapper. Returned table fields are 'lat', 'lon', 'tz', 'ok'.",
-		"(string):table");
+		"Given a street address or location name, returns latitude, longitude. Returns optional time zone if get_tz is true. Not designed to take latitude and longitude as input. Uses the MapQuest Geocoding API via a private NREL wrapper. Returned table fields are 'lat', 'lon', 'tz', 'ok'.",
+		"(string:location, [boolean:get_tz]):table");
 
-	double lat = 0, lon = 0, tz = 0;
-	// use GeoTools::GeocodeGoogle for non-NREL builds and set google_api_key in private.h
-	bool ok = GeoTools::GeocodeDeveloper(cxt.arg(0).as_string(), &lat, &lon, &tz);
+	bool get_tz = false;
+	if (cxt.arg_count() > 1) {
+		get_tz = cxt.arg(1).as_boolean();
+	}
+
+	double lat=NULL, lon=NULL, tz=NULL;
+	bool ok = false;
 	cxt.result().empty_hash();
+
+	// use GeoTools::GeocodeGoogle for non-NREL builds and set google_api_key in private.h
+ 	if (get_tz) {
+		ok = GeoTools::GeocodeDeveloper(cxt.arg(0).as_string(), &lat, &lon, &tz);
+		cxt.result().hash_item("tz").assign(tz); // converts NULL to 0
```

Comment:
Do we want to return 0 if the tz fails (no key)? tz=0 is valid for Greenwich.

No Azure key results
<img width="1752" height="1621" alt="image" src="https://github.com/user-attachments/assets/c84b60c3-5627-46ad-a2b5-4ef2db239fda" />

Reply:
TZ should be null or nan when it is not valid, thanks.